native/src/boot/bootimg.hpp

```diff
@@ -618,6 +622,12 @@ struct boot_img {
     uint8_t *extra;
     uint8_t *recovery_dtbo;
     uint8_t *dtb;
+    uint8_t *vendor_ramdisk_table;
+    uint8_t *bootconfig;
+
+    std::vector<std::pair<std::shared_ptr<struct vendor_ramdisk_table_entry_v4>, format_t>> ramdisk_table_entries;
```

Comment:
why `shared_ptr`

Reply:
It plays a bit nicer with vector. For example in unpack vector is accessed one time instead of four times, which makes the code more clear. Also for for-each isn't possible in that functions because I need i to generate the filenames.

Reply:
No a valid reason. As long as the ownership is shared, you should always use unique_ptr.

Reply:
Done

Reply:
I did my own modification. Please check if it works.

Reply:
Thanks, works fine.